libs/iota-browser/src/helpers/channel-provider.ts

```diff
@@ -176,32 +208,20 @@ export class ChannelProvider {
             const raw_data = toUtf8(
               messageReceivedEvent.message.payload as Buffer,
             )
+            let event
             try {
-              const event = JSON.parse(raw_data)
-              if (
-                event.eventType === 'signedRequest' &&
-                correlationId === event.correlationId
-              ) {
-                // TODO handle Zod errors gracefully
-                const signedRequest = SignedRequestEventSchema.parse(event)
-                const claims = jose.decodeJwt(signedRequest.data.jwt)
-                // TODO Zod validate JWT
-                if (!claims.client_id && typeof claims.client_id === 'string') {
-                  reject(new Error('Unexpected request claims received'))
-                }
-                const client_id = claims.client_id as string
-                const request: IotaChannelRequest = {
-                  correlationId: signedRequest.correlationId,
-                  payload: {
-                    request: signedRequest.data.jwt,
-                    client_id,
-                  },
-                }
-
-                resolve(request)
-              }
-            } catch (error) {
-              reject(error)
+              event = JSON.parse(raw_data)
+            } catch (e) {
+              reject(e)
+            }
+            if (correlationId !== event.correlationId) {
+              return
+            }
+            if (event.eventType === EventTypes.SignedRequest) {
+              const request = this.getRequest(event)
+              resolve(request)
+            } else if (event.eventType === EventTypes.Error) {
+              getError(event)
```

Comment:
we need to reduce the cognitive complexity here

Reply:
added a research ticket - FTL-15324